### ISSUE TITLE
Vehicle Body 2D

### ISSUE THREAD

**Issue by iapps207** [godotengine/godot-proposals#880]
**Describe the project you are working on:**
A car racing game 2d

**Describe the problem or limitation you are having in your project:**
No way to simulate a good vehicle behaviour in side view 2d mode  the car seems to hop unecessarily
![Vehicle Problem](https://user-images.githubusercontent.com/59492619/82751456-7890d880-9dc8-11ea-8ba0-a630d7aaeecc.PNG)

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
Adding a vehicle body 2d node thats helps everyone to simulate very good vehicle physics

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
The vehicle body 2d node can be ported from the 3d version node but this time simulating physics in the x and y axis only
**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
I tried that asked a lot of questions from pros they seem not to have an answer, So no it cant be worked around with a few lines of code

**Is there a reason why this should be core and not an add-on in the asset library?:**
There is already a vehicle body 3d node which is core and can be back ported for 2d

**Comment by Zireael07** [comment]
Note that there have been increasing noises to have vehicle body moved out of core.
And nope, I don't think it's as easy as porting the 3d version, because the 3d version is based on Bullet Vehicle physics, and our 2d physics aren't even Box2D, but totally custom.

**Comment by iapps207** [comment]
The vehicle bodies can be moved out of the core as long as there is a Vehicle body for both 2d and 3d because this method of using pinjoints is really tedious and dosent just give the right vehicle behaviour in 2d mode @Zireael07

**Comment by arkology** [comment]
> No way to simulate a good vehicle behaviour in side view 2d mode the car seems to jump unecessarily and then the pinjoint used for the tyres start to behave abnormally

Or maybe there are bugs with 2d physics in Godot.

**Comment by MCrafterzz** [comment]
I would really like a top-down 2d vehicle body, but that should probably be separate.

**Comment by Zireael07** [comment]
That was a little experiment and AFAIK it should be compatible with all 3.x versions of Godot.

**Comment by MCrafterzz** [comment]
> That was a little experiment and AFAIK it should be compatible with all 3.x versions of Godot.

Yes it does work in 3.2 and 4.0 but still it isn't something that is actively inproved
